flink-ml-lib/src/test/java/org/apache/flink/ml/recommendation/SwingTest.java

```diff
@@ -0,0 +1,236 @@
+
+        swing.setItemCol("item_1")
+                .setUserCol("user_1")
+                .setK(20)
+                .setMaxUserNumPerItem(500)
+                .setMinUserBehavior(10)
+                .setMaxUserBehavior(50)
+                .setAlpha1(5)
+                .setAlpha2(1)
+                .setBeta(0.35);
+
+        assertEquals("item_1", swing.getItemCol());
+        assertEquals("user_1", swing.getUserCol());
+        assertEquals(20, swing.getK());
+        assertEquals(500, swing.getMaxUserNumPerItem());
+        assertEquals(10, swing.getMinUserBehavior());
+        assertEquals(50, swing.getMaxUserBehavior());
+        assertEquals(5, swing.getAlpha1());
+        assertEquals(1, swing.getAlpha2());
+        assertEquals(0.35, swing.getBeta(), 1e-9);
+    }
+
+    @Test
+    public void testDataType() {
+        String msg = "";
+        List<Row> rows =
+                new ArrayList<>(Arrays.asList(Row.of(0, "10"), Row.of(1, "11"), Row.of(2, "")));
+
+        DataStream<Row> dataStream =
+                env.fromCollection(
+                        rows,
+                        new RowTypeInfo(
+                                new TypeInformation[] {
+                                    BasicTypeInfo.LONG_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO
+                                },
+                                new String[] {"user_id", "item_id"}));
+        Table data = tEnv.fromDataStream(dataStream);
+
+        try {
+            Table[] swingResultTables =
+                    new Swing()
+                            .setItemCol("item_id")
+                            .setUserCol("user_id")
+                            .setOutputCol("item_score")
+                            .setMinUserBehavior(1)
+                            .transform(data);
+            swingResultTables[0].execute().print();
+            fail();
+        } catch (RuntimeException e) {
+            msg = e.getMessage();
+            assertEquals("The types of user and item columns must be Long.", msg);
+        }
+    }
+
+    @Test
+    public void testNumberFormat() {
+        String msg = "";
+        List<Row> rows =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(0L, 10L),
+                                Row.of(null, 12L),
+                                Row.of(1L, 13L),
+                                Row.of(3L, 12L)));
+
+        DataStream<Row> dataStream = env.fromCollection(rows, trainDataTypeInfo);
+        Table data = tEnv.fromDataStream(dataStream);
+        Swing swing = new Swing().setItemCol("item_id").setUserCol("user_id").setMinUserBehavior(1);
+        Table[] swingResultTables = swing.transform(data);
+
+        try {
+            swingResultTables[0].execute().print();
+            fail();
+        } catch (RuntimeException e) {
+            msg = e.getCause().getCause().getCause().getCause().getCause().getMessage();
```

Comment:
We can replace this line with `ExceptionUtils.getRootCause`. You can check its usage in other tests.